pkg/checks/traceroute/traceroute_test.go

```diff
@@ -0,0 +1,149 @@
+package traceroute
+
+import (
+	"context"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/aeden/traceroute"
+	"github.com/caas-team/sparrow/pkg/checks"
+)
+
+func TestCheck(t *testing.T) {
+	type want struct {
+		expected map[string]Result
+	}
+	type testcase struct {
+		name string
+		c    *Traceroute
+		want want
+	}
+
+	cases := []testcase{
```

Comment:
Please consider refactoring the table driven tests to be aligned with other tests of the repository.

I know many find this approach to table driven tests better but imo it's less readable than just keeping everything in one unnamed struct.

Reply:
I think you misunderstood me. If we used an anonymous struct literal like this we wouldn't need to type out the struct for every test case:
```go
tests := []struct{
	name string
	c *Traceroute
	want map[string]Result
}{
	{
		name: "Success 5 hops",
		c:    newForTest(success(5), []string{"8.8.8.8"}),
		want: map[string]Result{
			"8.8.8.8": {
				NumHops: 5,
				Hops: []Hop{
					{Addr: "0.0.0.0", Latency: 0 * time.Second, Success: false},
					{Addr: "0.0.0.1", Latency: 1 * time.Second, Success: false},
					{Addr: "0.0.0.2", Latency: 2 * time.Second, Success: false},
					{Addr: "0.0.0.3", Latency: 3 * time.Second, Success: false},
					{Addr: "google-public-dns-a.google.com", Latency: 69 * time.Second, Success: true},
				},
			},
		},
	},
	{
		name: "Traceroute internal error fails silently",
		c:    newForTest(returnError(&net.DNSError{Err: "no such host", Name: "google.com", IsNotFound: true}), []string{"google.com"}),
		want: map[string]Result{
				"google.com": {Hops: []Hop{}},
		},
	},
}
```

With test cases like this we facilitate easier maintenance and readability. It even reduces the amount of boilerplate code (29 vs. 38 in your version). It's also more readable since you don't need the cognitive load of defining a `want` struct which is only an abstraction of defining the field directly. 

Named structs would only reduce boilerplate code if we used these structs also in other test functions.
Refer to this article for more details on table driven tests: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
I prefer creating a local type for this, since you can't always omit the type definition, like [here](https://github.com/caas-team/sparrow/blob/main/pkg/api/api_test.go#L42-L54) for example ![image](https://github.com/caas-team/sparrow/assets/27763017/3a1188f7-8d9c-4832-85b3-0ae662798f18)

Reply:
I agree with Tom on this one. Defining a type `want` to only hold an `expected` field, which is a simple `map[string]result` seems unnecessary. Omitting types in a `want` or `expected` field in a table driven test doesn't seem like a great idea as well - the test should express clear intention. 

go is a language of boilerplate and `if err!= nil`. It the language we love, it's ok to have to type something more. Copilot takes care of the most of it anyway :P

Reply:
In your example, I would go for wantMethod, wantPath, wantStatus. If this pattern repeats in multiple tests, then I would go for a refactored `want` struct. The `want` struct is IMO an abstraction which desires to cover everything, which is totally impossible, if more that one thing is expected. 

I think it's better to be verbose in tests, if we don't totally sacrifice maintainability, than to opt for abstractions which just reduce some boilerplate. Each test can and should look different, there's no need for some universal pattern (we already have that with table-driven sets and behavior-driven Testing.